sdk/src/main/java/com/mapbox/maps/renderer/MapboxRenderThread.kt

```diff
@@ -120,23 +123,30 @@ internal class MapboxRenderThread : Choreographer.FrameCallback {
     }
     if (!surface.isValid) {
       Logger.w(TAG, "EGL was configured but surface is not valid.")
-      postPrepareRenderFrame()
+      // give system a bit of time and try rendering again hoping surface will be valid now
+      postPrepareRenderFrame(delayMillis = RETRY_DELAY_MS)
       return false
     }
     // on Android SDK <= 23 at least on x86 emulators we need to force set EGL10.EGL_NO_CONTEXT
     // when resuming activity
     if (creatingSurface) {
       eglCore.makeNothingCurrent()
     }
-    eglSurface = eglCore.createWindowSurface(surface)
-    if (!eglCore.eglStatusSuccess) {
-      // Set EGL Surface as EGL_NO_SURFACE and try recreate it in next iteration.
-      eglSurface = EGL10.EGL_NO_SURFACE
-      postPrepareRenderFrame()
-      return false
+    if (eglSurface == null || eglSurface == EGL10.EGL_NO_SURFACE) {
```

Comment:
nit : maybe we don't need nullability since, let EglCore return EGL10.EGL_NO_SURFACE in case of errors? WIll simplify checks here

Reply:
we can not do that as we can't mock properly EGLSurface in tests then

Reply:
Seems tests could be tweaked to mock the necessary static fields, e.g. 
```
    mockkStatic(EGL10::class)
    val eglSurface = mockk<EGLSurface>()
    every { EGL10.EGL_NO_SURFACE } returns eglSurface
    Assert.assertEquals(mapboxRenderThread.eglSurface, eglSurface)
```
instead of checking `mapboxRenderThread.eglSurface` for null

Reply:
Ok, I'm wrong here, mockk doesn't allow java static fields to be mocked sadly. There's a hack with `mockkObject` which still doesn't work though